source/tutorials/packaging/contributing-to-nixpkgs.md

```diff
@@ -0,0 +1,222 @@
+    homepage = "https://github.com/atextor/icat";
+    license = licenses.bsdOriginal;
+    platforms = platforms.unix;
+  };
+...
+```
+
+With no further technical changes necessary, this derivation is now complete.
+
+## Adding to Nixpkgs
+You can now clone a local copy of Nixpkgs, and move your derivation into an appropriate location within that source tree.
+
+```console
+$ git clone https://github.com/nixos/nixpkgs
+```
+
+`icat` is a console tool which also displays graphics, so there are a few places within Nixpkgs that you could reasonably move it.
+
+In the past, packages were contributed based on the category they might fit in.
+
+For example: there are currently two other similar tools already in Nixpkgs, both of which are stored in `pkgs/applications/graphics`, so that might have been a good home for `icat` too.
+
+However, [Nixpkgs now prefers organization of packages by the the two-letter prefix of their package name](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#readme) within the `pkgs/by-name/` directory, [when possible](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#limitations).
+
+This eliminates the need to decide upon a particular category if a package fits into multiple, and it also eliminates the requirement of [updating `all-packages.nix` when using the category-based hierarchy](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning), and since our case isn't bound by any naming limitations, let's use the two-letter prefix system:
+
+```console
+$ mkdir nixpkgs/pkgs/by-name/ic/icat
+```
+
+You should rename `icat.nix` when you copy it to its new home, to keep to the convention of package derivation files [being named `package.nix`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#example) when using the two-letter prefix system:
+
+```console
+$ mv icat.nix nixpkgs/pkgs/applications/graphics/icat/default.nix
+```
+
+:::{note}
+Other organization systems might specify a different standard name for the derivation root. For example, the [category hierarchy](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#category-hierarchy) system requires the standard name `default.nix`.
+
+Additionally, as alluded to above, other hierarchy systems might require additional steps.
+
+For example, when using the categorical hierarchy, the relative directory paths to all packages get referenced in `pkgs/top-level/all-packages.nix`, which you would also need to update, putting the new package next to the other similar tools:
+
+```diff
+# all-packages.nix
+...
++ icat = callPackage ../applications/graphics/icat { };
+
+  imgcat = callPackage ../applications/graphics/imgcat { };
+
+  img-cat = callPackage ../applications/graphics/img-cat { };
+...
+```
+
+:::{note}
+Attributes in `all-packages.nix` are arranged *approximately* in alphabetical order, either by attribute name or by the directory containing their corresponding `default.nix`.
+:::
+:::
```

Comment:
I don't think we should go into detail on the old category hierarchy. In almost all cases it's not necessary anymore. We can just link to this instead: https://github.com/NixOS/nixpkgs/tree/master/pkgs#category-hierarchy